swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -25,35 +25,33 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
 
     Task {
       do {
-        var token = options?["token"] as? String
-        let keychain = Keychain()
-        let tokenRef = await keychain.search()
-
-        if let token = token {
-          // 1. If we're passed a token, save it to keychain
-
-          // Apple recommends updating Keychain items in place if possible
-          // In reality this won't happen unless there's some kind of race condition
-          // because we would have deleted the item upon sign out.
-          if tokenRef != nil {
-            try await keychain.update(token: token)
-          } else {
-            try await keychain.add(token: token)
-          }
+        // The tunnel can come up without the app, so initialize the id here
+        // as well.
+        try await FirezoneId.createIfMissing()
+
+        var token: Token?
+
+        if let tokenString = options?["token"] as? String,
```

Comment:
Make a ctor on `Token` to remove the low-level parsing here from the high-level flow.